lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -174,23 +199,30 @@ def success_from(response)
         response["errorCode"] == 0
       end
 
-      def message_from(response)
+      def message_from(response, options, action)
         if empty?(response["errorMessage"]) || response["errorMessage"] == "[ ]"
-          response["data"]["DSC_COD_ACCION"]
+          "#{response["data"]["DSC_COD_ACCION"]}, #{options[:error_message]}" if action == 'refund'
+          response['data']['DSC_COD_ACCION']
```

Comment:
This will always return `response['data']['DSC_COD_ACCION']`. Give the modification on line 204, is seems like we don't want to always return `response['data']['DSC_COD_ACCION']`.